pkg/config/config_template.yaml

```diff
@@ -1144,11 +1144,24 @@ api_key:
   ##  * require - list of key or key/value strings - traces must have those tags in order to be sent to Datadog
   ##  * reject - list of key or key/value strings - traces with these tags are dropped by the Agent
   ## Note: Rules take into account the intersection of tags defined.
+  ##       When `filter_tags` and `filter_tags_regex` are used at the same time, all rules are united for filtering.
+  ##       In the case when rule in `filter_tags` and `filter_tags_regex` have the same key, rule from `filter_tags`
+  ##       have precendence over the `filter_tags_regex`.
```

Comment:
For `filter_tags`, my understanding is that a string like `foo` appearing in either require or reject will match tags like `foo`, `foo:bar`, `foo:bing`, and a string like `foo:bar` will only match the tag `foo:bar`.  The existing comments above describing that behavior are less than clear, and perhaps some cleanup could be helpful :)

But, I'm not sure what that means for `filter_tags_regex`.  How does a regex have a key?  Are the strings in `<LIST_OF_KEY_VALUE_REGEX_TAGS>` of the form `<regex>:<regex>`?

Reply:
How do you think, may be better to write like this:
```
## @param filter_tags - object - optional
  ## Defines rules by which to filter traces based on tags.
  ##  * require - list of key or key/value strings - traces must have those tags in order to be sent to Datadog
  ##  * reject - list of key or key/value strings - traces with these tags are dropped by the Agent
  ## Please note that:
  ##   1. Rules take into account the intersection of tags defined.
  ##   2. When `filter_tags` and `filter_tags_regex` are used at the same time, all rules are united for filtering.
  ##      In the case when rule in `filter_tags` and `filter_tags_regex` have the same key, rule from `filter_tags`
  ##      have precendence over the `filter_tags_regex`.
  ##
  ##      In the case of the following configuration:
  ##        filter_tags:
  ##          require: ["foo:bar"]
  ##        filter_tags_regex:
  ##          require: ["foo:^bar[0-9]{1}$"]
  ##      resulting combined rule will match `foo:bar` and not `foo:bar1`
  #
  # filter_tags:
  #     require: [<LIST_OF_KEY_VALUE_TAGS>]
  #     reject: [<LIST_OF_KEY_VALUE_TAGS>]

  ## @param filter_tags_regex - object - optional
  ## Defines rules by which to filter traces based on tags with regex pattern for tag values.
  ##  * require - list of key or key/value regex pattern strings - traces must have those tags in order to be sent to Datadog
  ##  * reject - list of key or key/value regex pattern strings - traces with these tags are dropped by the Agent
  ## Note: Rules take into account the intersection of tags defined.
  #
  # filter_tags_regex:
  #     require: [<LIST_OF_KEY_VALUE_REGEX_TAGS>]    # e.g. ["<key>:<regex>"]
  #     reject: [<LIST_OF_KEY_VALUE_REGEX_TAGS>]     # e.g. ["<key>:<regex>"]
```

Reply:
Much better!

Reply:
✅